testing/tests/DevExpress.ui.widgets.form/form.markup.tests.js

```diff
@@ -773,6 +776,197 @@ QUnit.module('Form', () => {
+                                    label: { visible: false },
+                                    editorType,
+                                }],
+                            },
+                        ],
+                    },
+                ],
+            });
+            const optionLabelVisible = 'items[0].items[0].items[0].label.visible';
+
+            $formContainer
+                .dxForm('instance')
+                .option(optionLabelVisible, true);
+
+            const $componentWrapper = $formContainer.find(`.dx-${componentName}`).parent();
+
+            assert.strictEqual($componentWrapper.hasClass(TOGGLE_CONTROLS_PADDING_CLASS), true);
+        });
+
+        test(`${editorType} should not have class after visibility change to false (labelLocation=top, label.alignment=left)`, function(assert) {
+            const $formContainer = $('#form').dxForm({
+                labelLocation: 'top',
+                items: [
+                    {
+                        itemType: 'group',
+                        items: [
+                            {
+                                itemType: 'group',
+                                items: [{
+                                    dataField: editorType,
+                                    label: { visible: true },
+                                    editorType,
+                                }],
+                            },
+                        ],
+                    },
+                ],
+            });
+            const optionLabelVisible = 'items[0].items[0].items[0].label.visible';
+
+            $formContainer
+                .dxForm('instance')
+                .option(optionLabelVisible, false);
+
+            const $componentWrapper = $formContainer.find(`.dx-${componentName}`).parent();
+
+            assert.strictEqual($componentWrapper.hasClass(TOGGLE_CONTROLS_PADDING_CLASS), false);
+        });
+
+        test(`${editorType} should not have class when label.visible=false`, function(assert) {
+            const $form = $('#form').dxForm({
+                labelLocation: 'top',
+                items: [
+                    {
+                        itemType: 'group',
+                        items: [
+                            {
+                                itemType: 'group',
+                                items: [{
+                                    dataField: editorType,
+                                    label: { visible: false },
+                                    editorType,
+                                }, ],
+                            },
+                        ],
+                    },
+                ],
+            });
+            const $componentWrapper = $form.find(`.dx-${componentName}`).parent();
+
+            assert.strictEqual($componentWrapper.hasClass(TOGGLE_CONTROLS_PADDING_CLASS), false);
+        });
+
+        ['left', 'right'].forEach(labelLocation => {
```

Comment:
let's add all cases we think about in the code
```suggestion
        ['left', 'right', undefined].forEach(labelLocation => {
```